src/Microsoft.Data.Analysis.IO/DataFrameIO.cs

```diff
@@ -183,29 +89,125 @@ private static DataFrameColumn CreateColumn(Type kind, string[] columnNames, int
         /// </summary>
-        /// <param name="csvStream">stream of CSV data to be read in</param>
+        /// <param name="filename">filename</param>
         /// <param name="separator">column separator</param>
         /// <param name="header">has a header or not</param>
         /// <param name="columnNames">column names (can be empty)</param>
         /// <param name="dataTypes">column types (can be empty)</param>
-        /// <param name="numberOfRowsToRead">number of rows to read not including the header(if present)</param>
+        /// <param name="numRows">number of rows to read</param>
         /// <param name="guessRows">number of rows used to guess types</param>
         /// <param name="addIndexColumn">add one column with the row index</param>
         /// <param name="encoding">The character encoding. Defaults to UTF8 if not specified</param>
-        /// <returns><see cref="DataFrame"/></returns>
-        public static DataFrame LoadCsv(Stream csvStream,
+        /// <param name="cultureInfo">culture info for processing floats/doubles</param>
+        /// <returns>DataFrame</returns>
+        public static DataFrame LoadCsv(string filename,
+                                char separator = ',', bool header = true,
+                                string[] columnNames = null, Type[] dataTypes = null,
+                                int numRows = -1, int guessRows = 10,
+                                bool addIndexColumn = false, Encoding encoding = null,
+                                CultureInfo cultureInfo = null)
+        {
+            using (Stream fileStream = new FileStream(filename, FileMode.Open))
+            {
+                return LoadCsv(() => fileStream, SetTextFieldParserProperties,
+                                  separator: separator, header: header, columnNames: columnNames, dataTypes: dataTypes, numberOfRowsToRead: numRows,
+                                  guessRows: guessRows, addIndexColumn: addIndexColumn, encoding: encoding, cultureInfo: cultureInfo);
+                ;
+            }
+        }
+        public static DataFrame LoadCsv(Func<Stream> csvStream,
+                                Action<TextFieldParser> SetTextFieldParserProperties,
                                 char separator = ',', bool header = true,
                                 string[] columnNames = null, Type[] dataTypes = null,
                                 long numberOfRowsToRead = -1, int guessRows = 10, bool addIndexColumn = false,
-                                Encoding encoding = null)
+                                Encoding encoding = null, CultureInfo cultureInfo = null)
         {
-            if (!csvStream.CanSeek)
+            // if we have a comma separator and the cultureInfo has not been specified by the user, 
+            // we set it to Invariant Culture (since logically floats/doubles wouldn't be represented with commas in this case)
+            if(separator.Equals(',') && cultureInfo is null)
```

Comment:
In the case where the `CultureInfo` object is not null and the file has a comma based separator, we change the CultureInfo to Invariant in order to handle Double/Single values.

Reply:
So, for now I think this is ok since it's being more restrictive. I'm wondering if we can lose the `separator.Equals(',')` part of the if condition safely, but we can do that later too.